Sources/TensorFlow/Operators/Math.swift

```diff
@@ -2433,6 +2433,28 @@ internal extension Tensor where Scalar: TensorFlowFloatingPoint {
     }
 }
 
+extension Tensor {
```

Comment:
Currently, this extension is declared at an arbitrary location in `Sources/TensorFlow/Operators/Math.swift`.
Could you please move it to the bottom of `Sources/TensorFlow/Operators/Basic.swift`?

Please also add a header comment to the extension:
```swift
//===------------------------------------------------------------------------------------------===//
// Precondition utilities
//===------------------------------------------------------------------------------------------===//

extension Tensor {
    /// Returns `true` if the given axis is in the range `[-rank, rank)`.
    @usableFromInline
    internal func isAxisInRange<T: BinaryInteger>(_ axis: T) -> Bool {
        let axis = Int(axis)
        return axis >= -rank && axis < rank
    }

    /// Returns `true` if all given axes are in the range `[-rank, rank)`.
    @usableFromInline
    internal func areAxesInRange<T: BinaryInteger>(_ axes: [T]) -> Bool {
        return !axes.contains(where: { !isAxisInRange($0) })
    }

    /// Returns `true` if all scalars of the given 1-D tensor are in the range `[-rank, rank)`.
    @usableFromInline
    internal func areAxesInRange(_ axes: Tensor<Int32>) -> Bool {
        precondition(axes.rank == 1, "Axes must have rank 1")
        return areAxesInRange(axes.scalars)
    }
}
```